packages/mysql/lib/mysql_p.js

```diff
@@ -66,16 +66,42 @@ function patchCreatePool(mysql) {
   };
 }
 
+function patchGetConnection(pool) {
```

Comment:
Would this also work with `PoolCluster` if instead of grabbing the callback from the 1st argument, you grabbed it from the last argument?

https://github.com/mysqljs/mysql#poolcluster

Reply:
Looks like it would work, I'll add it into this PR, if you prefer me to split it up into a separate PR let me know (it may make the PR a bit big in size).